tests/logitech_receiver/test_profiles.py

```diff
@@ -0,0 +1,87 @@
+import re
+
+import logitech_receiver.hidpp20 as hidpp20
+import yaml
+
+
+def test_led_setting_bytes():
+    ebytes = bytes.fromhex("0A01020300500407000000")
+    setting = hidpp20.LEDEffectSetting.from_bytes(ebytes)
+    assert setting.ID == 0x0A, "LED setting ID read incorrectly"
+    assert setting.color == 0x010203, "LED setting color read incorrectly"
+    assert setting.period == 0x0050, "LED setting color read incorrectly"
+    assert setting.form == 0x04, "LED setting color read incorrectly"
+    assert setting.intensity == 0x07, "LED setting color read incorrectly"
+    bytes_out = setting.to_bytes()
+    assert ebytes == bytes_out, "LED setting bytes out not same as bytes in"
+
+
+def test_led_setting_yaml():
+    ebytes = bytes.fromhex("0A01020300500407000000")
+    eyaml = (
+        "!LEDEffectSetting {ID: !NamedInt {name: Breathe, value: 0xa}, color: 0x10203, "
+        "form: 0x4, intensity: 0x7, period: 0x50} "
+    )
+    setting = hidpp20.LEDEffectSetting.from_bytes(ebytes)
+    assert setting.ID == 0x0A, "LED setting ID read incorrectly"
+    assert setting.color == 0x010203, "LED setting color read incorrectly"
+    assert setting.period == 0x0050, "LED setting color read incorrectly"
+    assert setting.form == 0x04, "LED setting color read incorrectly"
+    assert setting.intensity == 0x07, "LED setting color read incorrectly"
+    yaml_out = yaml.dump(setting)
+    assert eyaml == re.compile(r"\s+").sub(" ", yaml_out)
+    setting = yaml.safe_load(eyaml)
+    assert setting.to_bytes() == ebytes
+
+
+def test_button_bytes():
```

Comment:
If parametrize is not well suitable for this test case, please improve the order a bit. A good approach for ordering code inside a test is

1. Setup and define expected values
2. Run code under test
3. Assert

and to split each of these groups with a new line.